src/fieldAnalyser/typeUtils.ts

```diff
@@ -0,0 +1,39 @@
+import {FieldTypes} from '@coveord/platform-client';
+
+const acceptedTransitions: Partial<Record<FieldTypes, FieldTypes[]>> = {
+  [FieldTypes.LONG]: [FieldTypes.LONG_64, FieldTypes.DOUBLE],
+  [FieldTypes.LONG_64]: [FieldTypes.DOUBLE],
+};
+
+export function isValidTypeTransition(
+  currentState: FieldTypes,
+  nextState: FieldTypes
+): boolean {
+  const sameStateTransition = currentState === nextState;
+  const differentStateTransition = (
+    acceptedTransitions[currentState] || []
+  ).includes(nextState);
+  return sameStateTransition || differentStateTransition;
+}
+
+export function getGuessedTypeFromValue(obj: unknown): FieldTypes {
+  switch (typeof obj) {
+    case 'number':
+      return getSpecificNumericType(obj);
+    case 'string':
+      return FieldTypes.STRING;
+    default:
+      return FieldTypes.STRING;
+  }
+}
+
+function getSpecificNumericType(number: number): FieldTypes {
+  const isInteger = (x: number): boolean => Math.floor(x) === x;
+  const is32BitInteger = (x: number) => (x | 0) === x;
```

Comment:
So we're at the point where we're doing bitwise operation.
This can only end well. 😂 
(tho, yeah, I mean, it work lol)